pkg/storage/db.go

```diff
@@ -211,11 +217,19 @@ func (db *DB) MigrateSchemaDown(ctx context.Context) (uint, error) {
 
 // MigrateSchemaToVersion performs one or more database migrations to bring schema to the specified version
 func (db *DB) MigrateSchemaToVersion(ctx context.Context, version uint) error {
+	err := db.sqldb.PingContext(ctx)
+	if err != nil {
+		return err
+	}
 	return db.migrator.Migrate(version)
 }
 
 // GetSchemaVersion retrieves the current version of database schema
 func (db *DB) GetSchemaVersion(ctx context.Context) (uint, error) {
+	err := db.sqldb.PingContext(ctx)
+	if err != nil {
+		return 0, err
+	}
```

Comment:
do we need this change in migrateSchema() func as well (line 245, called by MigrateSchemaUp/MigrateSchemaDown)?